akka-stream/src/main/scala/akka/stream/impl/Timers.scala

```diff
@@ -138,13 +138,21 @@ object Timers {
 
   }
 
-  final class IdleTimeoutBidi[I, O](val timeout: FiniteDuration) extends GraphStage[BidiShape[I, I, O, O]] {
+  /**
+   * @param timeoutToTermination allows customising the behaviour once a timeout triggers.
+   *                             Returning None causes completion, and Some(exception) causes failure of the stage with that exception.
+   */
+  final class IdleTimeoutBidi[I, O](val timeout: FiniteDuration, timeoutToTermination: () ⇒ Option[Throwable]) extends GraphStage[BidiShape[I, I, O, O]] {
```

Comment:
`recover` _would log_ on `ERROR` if I `throw` in it, which is why we'd need the `mapError`, because I don't want this class to log anything. I will do the logging, but I need a specific exception, not a random `TimeoutException` to give users a good experience.